File: LC_11_container_with_most_water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """

        if len(height) == 0:
            return -1
        
        low = 0
        high = len(height)-1
        currArea = -1
        


        while(low < high):
            result = min(height[low],height[high]) * (high-low)
            print(result)
            currArea = max(currArea,result) 

            if height[low]<=height[high]:
                low = low + 1
            else:
                high = high - 1

        return currArea



        
        
Console

File: LC11-containerWithMostWater.java
class Solution {
    public int maxArea(int[] height) {

        int l = 0;
        int r = height.length -1;
        int max = 0;
        while (l<r) {
            max = Math.max(max, Math.min(height[l],height[r])*(r-l));

            // move ptr of lower (because min will decide the water contain)
            if (height[l] <= height[r]) {
                l++;
            }
            else {
                r--;
            }
        }

        return max;
    }
}